### ISSUE TITLE
reference management in image layout

### ISSUE THREAD

**Issue by xiekeyang** [opencontainers/image-spec#374]
Current spec doesn't care the layout management for `ref` folder. Actually consumer can set up many child folder and put the reference file to any where. It seems to be a little chaos. I think it had better to build rules for this. like:

`<domain>/<repository>/<reference>`

`domain` identifies a directory.
`repository` identifies the image name, which may be multi-level directory, like `library/ubuntu`
`reference` identifies the image tag.

the tree looks like:

```
.
└── refs
    └── docker.registry.io
        └── library
            └── ubuntu
                └── latest
                └── 14.04
            └── busybox
                └── latest
```

Any `ref` layout inconsistent to above will be regarded as illegal.

Previous [comment](https://github.com/opencontainers/image-spec/pull/327#issuecomment-248256133) shows that the image layout actually play as an image repository, which contains many images(as it contains manifest list). So I think the directory should be managed with multi-levels.

**Comment by wking** [comment]
On Sat, Oct 08, 2016 at 01:10:57AM -0700, xiekeyang wrote:
> Current spec doesn't care the layout management for `ref`
> folder. Actually consumer can set up many child folder and put the
> reference file to any where. It seems to be a little chaos. I think
> it had better to build rules for this. like:
> 
> `<domain>/<repository>/<reference>`

This is not legal, because ‘/’ is not in the [A-Za-z0-9-._] whitelist
[1].  Previous discussion in #174 and #219.

[1]: https://github.com/opencontainers/image-spec/blame/v1.0.0-rc1/image-layout.md#L46

**Comment by stevvooe** [comment]
Closed by #438

**Comment by xiekeyang** [comment]
closed by #438
